cpp/examples/tracing_server.cpp

```diff
@@ -0,0 +1,143 @@
+    proton::connection conn_;
+    std::map<std::string, proton::sender> senders_;
+
+  public:
+    server(const std::string& u, const std::string& a) :
+        conn_url_(u), addr_(a) {}
+
+    void on_container_start(proton::container& c) override {
+        conn_ = c.connect(conn_url_);
+        conn_.open_receiver(addr_);
+
+        std::cout << "Server connected to " << conn_url_ << std::endl;
+    }
+
+    std::string to_upper(const std::string& s) {
+        std::string uc(s);
+        size_t l = uc.size();
+
+        for (size_t i=0; i<l; i++) {
+            uc[i] = static_cast<char>(std::toupper(uc[i]));
+        }
+
+        return uc;
+    }
+
+    void on_message(proton::delivery&, proton::message& m) override {
+
+        opentelemetry::nostd::shared_ptr<opentelemetry::trace::Tracer> tracer = provider->GetTracer("qpid-tracer", OPENTELEMETRY_SDK_VERSION);
+        opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span> s = tracer->StartSpan("server-on-message",
+            {{"Delivery_tag", "delivery_tag"}, {"Destination_address", "t_addr"}});
+
+        opentelemetry::trace::Scope sc = tracer->WithActiveSpan(s);
+
+        std::cout << "Received " << m.body() << std::endl;
+
+        std::string reply_to = m.reply_to();
+        proton::message reply;
+
+        reply.to(reply_to);
+        reply.body(to_upper(proton::get<std::string>(m.body())));
+        reply.correlation_id(m.correlation_id());
+
+        if (!senders_[reply_to]) {
+            senders_[reply_to] = conn_.open_sender(reply_to);
+        }
+
+        senders_[reply_to].send(reply);
+    }
+};
+
+int main(int argc, char** argv) {
+    try {
+        std::string conn_url = argc > 1 ? argv[1] : "//127.0.0.1:5672";
+        std::string addr = argc > 2 ? argv[2] : "examples";
+        std::string t = argc > 3 ? argv[3] : "doesnotmatter";
+
+        opentelemetry::exporter::jaeger::JaegerExporterOptions opts;
+
+        // Initialize Jaeger Exporter
+        std::unique_ptr<opentelemetry::sdk::trace::SpanExporter> exporter = std::unique_ptr<opentelemetry::sdk::trace::SpanExporter>(
+            new opentelemetry::exporter::jaeger::JaegerExporter(opts));
+        
+        auto resource_attributes = opentelemetry::sdk::resource::ResourceAttributes
+        {
+            {"service.name", "qpid-server"}
+        };
+
+        auto resource = opentelemetry::sdk::resource::Resource::Create(resource_attributes);
+
+        auto processor = std::unique_ptr<opentelemetry::sdk::trace::SpanProcessor>(
+            new opentelemetry::sdk::trace::SimpleSpanProcessor(std::move(exporter)));
+        provider = opentelemetry::nostd::shared_ptr<opentelemetry::trace::TracerProvider>(
+            new opentelemetry::sdk::trace::TracerProvider(std::move(processor), resource));
+
+        // Set the global trace provider
+        opentelemetry::trace::Provider::SetTracerProvider(provider);
+    
+        // Enable tracing in proton cpp
+        if (t == "tracing")
```

Comment:
Note: No longer need this line as this example is solely for demonstrating tracing in proton. Same for the tracing_client.